integration-tests/tests/singlestore/singlestore-common.ts

```diff
@@ -0,0 +1,3440 @@
+} from 'drizzle-orm/singlestore-core';
+import type { SingleStoreRemoteDatabase } from 'drizzle-orm/singlestore-proxy';
+import { migrate } from 'drizzle-orm/singlestore/migrator';
+import getPort from 'get-port';
+import { v4 as uuid } from 'uuid';
+import { afterAll, beforeEach, describe, expect, expectTypeOf, test } from 'vitest';
+import { Expect, toLocalDate } from '~/utils.ts';
+import type { Equal } from '~/utils.ts';
+
+type TestSingleStoreDB = SingleStoreDatabase<any, any>;
+
+declare module 'vitest' {
+	interface TestContext {
+		singlestore: {
+			db: TestSingleStoreDB;
+		};
+		singlestoreProxy: {
+			db: SingleStoreRemoteDatabase;
+		};
+	}
+}
+
+const ENABLE_LOGGING = false;
+
+const usersTable = singlestoreTable('userstest', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	verified: boolean('verified').notNull().default(false),
+	jsonb: json('jsonb').$type<string[]>(),
+	createdAt: timestamp('created_at', { fsp: 6 }).notNull().defaultNow(),
+});
+
+const users2Table = singlestoreTable('users2', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	cityId: int('city_id'),
+});
+
+const citiesTable = singlestoreTable('cities', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+});
+
+const usersOnUpdate = singlestoreTable('users_on_update', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	updateCounter: int('update_counter').default(sql`1`).$onUpdateFn(() => sql`update_counter + 1`),
+	updatedAt: datetime('updated_at', { mode: 'date', fsp: 6 }).$onUpdateFn(() => new Date()),
+	alwaysNull: text('always_null').$type<string | null>().$onUpdateFn(() => null), // need to add $type because $onUpdate add a default value
+});
+
+const datesTable = singlestoreTable('datestable', {
+	date: date('date'),
+	dateAsString: date('date_as_string', { mode: 'string' }),
+	time: time('time', { fsp: 1 }),
+	datetime: datetime('datetime', { fsp: 6 }),
+	datetimeAsString: datetime('datetime_as_string', { fsp: 6, mode: 'string' }),
+	timestamp: timestamp('timestamp', { fsp: 6 }),
+	timestampAsString: timestamp('timestamp_as_string', { fsp: 6, mode: 'string' }),
+	year: year('year'),
+});
+
+const coursesTable = singlestoreTable('courses', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	categoryId: int('category_id'),
+});
+
+const courseCategoriesTable = singlestoreTable('course_categories', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+});
+
+const orders = singlestoreTable('orders', {
+	id: serial('id').primaryKey(),
+	region: text('region').notNull(),
+	product: text('product').notNull().$default(() => 'random_string'),
+	amount: int('amount').notNull(),
+	quantity: int('quantity').notNull(),
+});
+
+const usersMigratorTable = singlestoreTable('users12', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	email: text('email').notNull(),
+}, (table) => {
+	return {
+		name: uniqueIndex('').on(table.name).using('btree'),
```

Comment:
nit: do we need to specify `btree` here? Isn't that the only mode we have for columnstore? And I guess related to that, should we disallow `using` for an index builder on columnstore? Not really something for this PR though

Reply:
Agree that we should remove `using` since we only have the one mode